plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/application/NewApplicationWizard.java

```diff
@@ -0,0 +1,92 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors: Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.wizard.application;
+
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jface.dialogs.ErrorDialog;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.jface.wizard.IWizardContainer;
+import org.eclipse.jface.wizard.Wizard;
+import org.eclipse.jface.wizard.WizardDialog;
+import org.eclipse.ui.IImportWizard;
+import org.eclipse.ui.INewWizard;
+import org.eclipse.ui.IWorkbench;
+import org.jboss.tools.common.ui.JobUtils;
+import org.jboss.tools.common.ui.WizardUtils;
+import org.jboss.tools.openshift.internal.common.core.job.AbstractDelegatingMonitorJob;
+import org.jboss.tools.openshift.internal.ui.OpenShiftUIActivator;
+import org.jboss.tools.openshift.internal.ui.job.CreateApplicationFromTemplateJob;
+
+/**
+ * The new application wizard that allows you to create an application given an
+ * OpenShift template
+ */
+public class NewApplicationWizard extends Wizard implements IImportWizard, INewWizard {
+	
+	private NewApplicationWizardModel model;
+
+	public NewApplicationWizard(NewApplicationWizardModel model) {
+		setWindowTitle("New OpenShift Application");
+		this.model = model;
+	}
+	
+	@Override
+	public void addPages() {
+		addPage(new TemplateListPage(this, model));
+		addPage(new TemplateParametersPage(this, model));
+		addPage(new ResourceLabelsPage(this, model));
+	}
+	
+	@Override
+	public void init(IWorkbench workbench, IStructuredSelection selection) {
+		
```

Comment:
@adietish I have not completely thought through the other places where this dialog could launch other then from the context menu, but we will need to know what project this template will be applied against.  There may be some other additional initialization required.  Users's will mostly likely only have authorization to add resources to projects where they have authorization.